### PR TITLE
[iOS] Playlist Revamp Improvements

### PR DESCRIPTION
**Pull request by kylehickinson**
This adds some changes based on feedback from design/product, as well as adds oEmbed thumbnail support

Disclaimer: This feature is still behind a feature flag, FIXMEs will be removed/cleaned up before the flag is flipped on by default

Resolves

## Submitter Checklist:

- [ ] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [ ] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

File: package.json
Comment (diracdeltas) [inline package.json:198]: needs a quick security review

Comment (kylehickinson) [reply package.json:198]: Yep! Opened one here already 🙂 https://github.com/brave/reviews/issues/1649

Comment (diracdeltas) [reply package.json:198]: it looks like we are not running any code from this npm package, just using a JSON file that it provides? can we just bundle https://oembed.com/providers.json and update it periodically as needed? that would avoid adding an additional dependency which could be compromised someday, add postinstall scripts, etc.

Comment (kylehickinson) [reply package.json:198]: We can certainly do that if we want. I mostly choose this route as it provided the easiest way to keep track of its updates, but yeah its just a single json file we're pulling in so if we want I can just drop the file into the iOS project instead.

Comment (kylehickinson) [reply package.json]: Pushed the change, providers json file is now just bundled with the app directly and no changes to package.json anymore!

File: ios/brave-ios/Sources/PlaylistUI/OEmbedThumbnailFetcher.swift
Comment (iccub) [inline ios/brave-ios/Sources/PlaylistUI/OEmbedThumbnailFetcher.swift:33]: Can we use logger here?

**Review by iccub — APPROVED** [review]
lgtm
let's make sure the security ticket is closed before we merge

File: ios/brave-ios/Sources/PlaylistUI/OEmbedThumbnailFetcher.swift
Comment (kylehickinson) [reply ios/brave-ios/Sources/PlaylistUI/OEmbedThumbnailFetcher.swift]: Done!

File: ios/brave-ios/Sources/PlaylistUI/OEmbedThumbnailFetcher.swift
Comment (stoletheminerals) [inline ios/brave-ios/Sources/PlaylistUI/OEmbedThumbnailFetcher.swift:92]: there is a lot of http urls in the json. how does HTTPS autoupgrade work with it?

Comment (kylehickinson) [reply ios/brave-ios/Sources/PlaylistUI/OEmbedThumbnailFetcher.swift:92]: It would not participate since none of this happens inside of the browser web views.

Comment (stoletheminerals) [reply ios/brave-ios/Sources/PlaylistUI/OEmbedThumbnailFetcher.swift:92]: When I go to http://animoto.com/ it redirects me to https://animoto.com/. If I add a video to playlist from there Brave will send the request to http, because we get the base domain from the url and look it up in the json, which has this:
```
{
        "provider_name": "Animoto",
        "provider_url": "http://animoto.com/",
        "endpoints": [
            {
                "schemes": [
                    "http://animoto.com/play/*"
                ],
                "url": "http://animoto.com/oembeds/create"
            }
        ]
    }
```

That doesn't look good to me.

Comment (stoletheminerals) [reply ios/brave-ios/Sources/PlaylistUI/OEmbedThumbnailFetcher.swift:92]: I'd say we should override the scheme and use the one from the tab

Comment (kylehickinson) [reply ios/brave-ios/Sources/PlaylistUI/OEmbedThumbnailFetcher.swift:92]: I mean if we want I can just force https for all oembed calls, if they fail there's always a fallback anyways so its not a huge deal

Comment (stoletheminerals) [reply ios/brave-ios/Sources/PlaylistUI/OEmbedThumbnailFetcher.swift:92]: Sounds good!

Comment (kylehickinson) [reply ios/brave-ios/Sources/PlaylistUI/OEmbedThumbnailFetcher.swift:93]: Added explicit scheme set here: https://github.com/brave/brave-core/pull/24163/commits/3f807622be01360119091a080b836183d93e801c#diff-88e28104cda111e833c2efb99205b1d4b07ef428c1a294e0e8083dfc19ab7959R80

**Review by stoletheminerals — APPROVED** [review]

**Pull request merged**
